app/main.py

```diff
@@ -1,8 +1,40 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int):
+        self.name = name
+        self.age = age
+
+        Person.people[name] = self
+
+    def set_spouse(self, spouse_name):
+
+        if spouse_name is None:
+            return
+
+        spouse = Person.people.get(spouse_name)
+        if spouse:
+            if not hasattr(self, "wife") and not hasattr(spouse, "husband"):
+                self.wife = spouse
+                spouse.husband = self
+            elif not hasattr(self, "husband") and not hasattr(spouse, "wife"):
+                self.husband = spouse
+                spouse.wife = self
```

Comment:
The logic for setting spouses assumes that a person can only have one spouse of the opposite gender. If the task requirements allow for more complex relationships, this logic might need to be adjusted. Additionally, consider adding checks to ensure that a person is not setting themselves as their own spouse.